### PR TITLE
[JUJU-501] Remove orphaned charm documents

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following updates the charmhub refresh API clients so that all
results are correctly ordered. This was written in the documentation of
the charmhub client API, but it was not observed when using the client.
This can be seen using the charmrevisionupdater expects all results to
be in order. If the results are out of order we end up creating place
holders with invalid revisions.

These are valid:

```
 - ch:amd64/focal/mongodb-64: {}
 - ch:amd64/focal/ubuntu-19: {}
```

After running they become:

```
 - ch:amd64/focal/mongodb-19: {}
 - ch:amd64/focal/mongodb-64: {}
 - ch:amd64/focal/ubuntu-19: {}
 - ch:amd64/focal/ubuntu-64: {}
```

## QA steps

```sh
$ juju bootstrap lxd test
$ juju enable-ha
$ juju deploy ubuntu
$ juju deploy mongodb
$ juju ssh -m controller 0
$ juju_engine_report | grep "ch:amd64/focal/"
```
Outputs:
```
          ch:amd64/focal/mongodb-64: {}	
          ch:amd64/focal/ubuntu-19: {}
```

Step the mongo primary down:
```sh
$ juju mongo
$ rs.stepDown(10)
```

```sh
$ juju ssh -m controller 0
$ juju_engine_report | grep "ch:amd64/focal/"
```

Invalid output:
```sh
          ch:amd64/focal/mongodb-19: {}
          ch:amd64/focal/mongodb-64: {}
          ch:amd64/focal/ubuntu-19: {}
          ch:amd64/focal/ubuntu-64: {}
```

The solution is to sort all results from the client.

Result should be:
```
          ch:amd64/focal/mongodb-64: {}	
          ch:amd64/focal/ubuntu-19: {}
```

### CODE REVIEW

File: charmhub/refreshconfig.go [hunk 1]
```diff
@@ -91,9 +91,8 @@ type executeOne struct {
 	Base     RefreshBase
 	// instanceKey is a private unique key that we construct for CharmHub API
 	// asynchronous calls.
-	action            Action
-	instanceKey       string
-	resourceRevisions []transport.RefreshResourceRevision
```
Comment (SimonRichardson) [inline charmhub/refreshconfig.go:96]: driveby: removed because it wasn't used.

**Review by hmlanigan — APPROVED** [review]
The upgrade step works well, I added an older charm revision to the model and the placeholder for the new version was not removed.  `juju refresh` was successful after

File: upgrades/operations.go [hunk 2]
```diff
@@ -57,6 +57,7 @@ var stateUpgradeOperations = func() []Operation {
 		upgradeToVersion{version.MustParse("2.9.20"), stateStepsFor2920()},
 		upgradeToVersion{version.MustParse("2.9.22"), stateStepsFor2922()},
 		upgradeToVersion{version.MustParse("2.9.24"), stateStepsFor2924()},
+		upgradeToVersion{version.MustParse("2.9.25"), stateStepsFor2925()},
```
[the remark this answers is not in the document]
Comment (SimonRichardson) [reply upgrades/operations.go:60]: The 2.9.24 release is stuck in CI limbo. Officially it should be with SQA as a candidate and 2.9.25 is the next release.

**Comment by SimonRichardson** [comment]
$$merge$$
